packages/drivers/odsp-driver/src/snapshotFormatConverter.ts

```diff
@@ -0,0 +1,279 @@
+            snapshotSeqNumber: node.getNumber(5),
+        };
+        assert(minReadVersion >= header.minReadVersion, "Driver min read version should >= to server minReadVersion");
+        assert(header.createVersion >= minReadVersion, "Snapshot should be created with minReadVersion or above");
+    }
+
+    /**
+     * Recreates dictionary section of the tree.
+     * @param node - tree node to read dictionary section from. This container paths/ids mapping
+     * to integer representation.
+     */
+    private readDictionarySection(node: NodeCore) {
+        const dictionary = new Array<string>();
+        for (const name of node) {
+            assert(name instanceof BlobCore, "Mapping should be of type BlobCore");
+            dictionary.push(name.toString());
+        }
+        return dictionary;
+    }
+
+    /**
+     * Recreates blobs section of the tree.
+     * @param node - tree node to read blob section from
+     * @param dictionary - name map, used to decode path/IDs.
+     */
+    private readBlobSection(node: NodeCore, dictionary: string[]) {
+        const blobs: Map<string, Uint8Array> = new Map();
+        for (const [idIndex, blob] of node.iteratePairs()) {
+            assert(typeof idIndex === "number", "Blob index should be a number");
+            assert(blob instanceof BlobCore, "Blob content should be of type blob");
+            const blobId = dictionary[idIndex];
+            assert(blobId !== undefined, "blob id should be present");
+            blobs.set(blobId, blob.buffer);
+        }
+        return blobs;
+    }
+
+    /**
+     * Recreates ops section of the tree.
+     * @param node - tree node to read ops section from
+     */
+    private readOpsSection(node: NodeCore) {
+        const ops: ISequencedDeltaOpMessage[] = [];
+        for (let i = 0; i < node.length; ++i) {
+            ops.push(JSON.parse(node.getString(i)));
+        }
+        return ops;
+    }
+
+    /**
+     * Recreates snapshot tree out of tree representation.
+     * @param treeNode - tree node to de-serialize from
+     * @param dictionary - name map, used to decode path/IDs.
+     */
+    private readTreeSection(treeNode: NodeCore, dictionary: string[]) {
+        const tree: ISnapshotTree = {
+            id: "id",
+            blobs: {},
+            commits: {},
+            trees: {},
+        };
+        for (const [pathIndex, child] of treeNode.iteratePairs()) {
+            assert(typeof pathIndex == "number", "Tree index should be a number");
+            const path = dictionary[pathIndex];
+            assert(path !== undefined, "Path should not be undefined");
+
+            if (child instanceof NodeCore) {
+                tree.trees[path] = this.readTreeSection(child, dictionary);
+            } else {
+                assert(typeof child == "number", "Should be number to look in dictionary");
+                const id = dictionary[child];
+                assert(id !== undefined, "Blob id should not be undefined");
```

Comment:
I think the right message here - id is out of range, as we get undefined if we read outside of range of array in JS